### PR TITLE
docs(auth-provider): update descriptions and refer to tutorial

### PR DESCRIPTION
**Pull request by alicanerdurmaz**
added: https://refine.dev/docs/api-reference/core/providers/auth-provider

### CODE REVIEW

**Comment by alicanerdurmaz** [comment]
Should we add code snippets to the methods section?

File: documentation/docs/api-reference/core/providers/auth-provider.md [hunk 1]
```diff
@@ -4,45 +4,36 @@ title: Auth Provider
...
+-   **Strapi** &#8594 [Source Code](https://github.com/refinedev/refine/tree/master/examples/data-provider-strapi-v4) - [Demo](/docs/examples/data-provider/strapi-v4)
+-   **Strapi GraphQL** &#8594 [Source Code](https://github.com/refinedev/refine/tree/master/examples/data-provider-strapi-graphql) - [Demo](/docs/examples/data-provider/strapi-graphql)
+-   **Nhost** &#8594 [Source Code](https://github.com/refinedev/refine/tree/master/examples/data-provider-nhost) - [Demo](/docs/examples/data-provider/nhost)
+-   **Basic with Nextjs** &#8594 [Source Code](https://github.com/refinedev/refine/tree/master/examples/with-nextjs) - [Demo](/docs/examples/next-js/)
+-   **Basic with Remix** &#8594 [Source Code](https://github.com/refinedev/refine/tree/master/examples/with-remix-headless) - [Demo](/docs/examples/remix/remix-headless)
 
 :::
 
-## Usage
+## Basic Usage
```
Comment (salihozdemir) [inline documentation/docs/api-reference/core/providers/auth-provider.md:36]:
```suggestion
## Using Auth Providers in refine
```
In the tutorial, we wrote like this. Is it good? I'm not sure.

**Review by salihozdemir — APPROVED** [review]

File: documentation/docs/api-reference/core/providers/auth-provider.md [hunk 2]
```diff
@@ -68,834 +59,168 @@ const App = () => {
...
-   ...
+    login: () => Promise.resolve(),
+    register: () => Promise.resolve(),
+    forgotPassword: () => Promise.resolve(),
+    updatePassword: () => Promise.resolve(),
+    logout: () => Promise.resolve(),
+    checkAuth: () => Promise.resolve(),
+    checkError: () => Promise.resolve(),
+    getPermissions: () => Promise.resolve(),
+    getUserIdentity: () => Promise.resolve(),
```
[the remark this answers is not in the document]
Comment (alicanerdurmaz) [reply documentation/docs/api-reference/core/providers/auth-provider.md:68]: But, This is an `object` not an `interface`. Should we change to `interface`?

File: documentation/docs/api-reference/core/providers/auth-provider.md [hunk 3]
```diff
@@ -68,834 +59,168 @@ const App = () => {
...
 };
 ```
 
-<br />
+### login <PropTag required />
```
[the remark this answers is not in the document]
Comment (yildirayunlu) [reply documentation/docs/api-reference/core/providers/auth-provider.md:78]: I think its not necessary.

File: documentation/docs/api-reference/core/providers/auth-provider.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (yildirayunlu) [reply documentation/docs/api-reference/core/providers/auth-provider.md:65]: I think, can be separated with a comment.

**Pull request merged**